bills/serializers.py

```diff
@@ -0,0 +1,58 @@
+# pylint: disable=too-few-public-methods,missing-class-docstring
+"""
+Serializers for bills application.
+Id fields left to facilitate frontend work.
+"""
+
+from rest_framework.serializers import HyperlinkedIdentityField, ModelSerializer
+
+from .models import Bill, Event, Participant, Payment
+
+
+class EventSerializer(ModelSerializer):
+    """Serializer for Event object"""
+
+    url = HyperlinkedIdentityField(view_name="event-detail")
+    participants_url = HyperlinkedIdentityField(
+        view_name="event-participants-list", lookup_url_kwarg="event_pk"
+    )
+    bills_url = HyperlinkedIdentityField(
+        view_name="event-bills-list", lookup_url_kwarg="event_pk"
+    )
+    payments_url = HyperlinkedIdentityField(
+        view_name="event-payments-list", lookup_url_kwarg="event_pk"
+    )
+
+    class Meta:
+        model = Event
+        fields = "__all__"
+
+
+class ParticipantSerializer(ModelSerializer):
+    """Serializer for Participant object"""
+
+    url = HyperlinkedIdentityField(view_name="participant-detail")
+
+    class Meta:
+        model = Participant
+        fields = "__all__"
```

Comment:
This serializer is used in nested view, under event endpoint, so you should exclude `event` field. API should automatically fetch event from url path when creating new participant.

Reply:
You can use `HiddenField` for that:
```from rest_framework.fields import HiddenField

class ParticipantSerializer(ModelSerializer):
    """Serializer for Participant object"""

    url = HyperlinkedIdentityField(view_name="participant-detail")
    event = HiddenField(default=None)

    class Meta:
        model = Participant
        fields = "__all__"

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.fields["event"].default = self.view.get_event()
```

Don't forget to implement `get_event` method in viewset

```
from rest_framework.generics import get_object_or_404

def get_queryset(self):
    """Query of all Participant objects being related to the given Event"""

    return model.objects.filter(event=self.kwargs["event_pk"])

def get_event(self):
    return get_object_or_404(self.get_queryset())
```

Reply:
With `get_event` method `EventRelationshipsViewset` class may become too complicated. Please consider to use separate viewset class for events and use `create_viewset` generator only for nested views, so you could get rid of `if nested`

Reply:
Done. It wasn't that easy :)